### PR TITLE
DRAFT: Transition django-registration to django-allauth for future SSO work

### PR DESCRIPTION
**Pull request by bufke**
Related to https://github.com/kobotoolbox/kpi/issues/3775
:exclamation:  Merge #4102 first

## Description

Migrate to django-allauth accounts framework. Support email verification when changing an email. Require POST for logout.

## Implementation Details

django-allauth was deemed the best option for it's support of many oauth clients. Drawbacks include a lack of true multi-tenant configuration and lack of proper OIDC support. We could add both features in theory, ourselves.

- Port django registration templates/logic to allauth
- Make logout require a POST (best practice) frontend was modified to support this
- Add Microsoft Azure Active Directory authentication via django-allauth package but disabled by default

### CODE REVIEW

**Comment by paulschreiber** [comment]
See also work we have done in https://github.com/techmatters/kpi/tree/feat-terraso-auth

**Comment by bufke** [comment]
Things for jnm to review

- General impact of the registration to allauth migration
- Email verification migration that assumes any existing email is verified and marks them for potential later intervention
- Account settings chosen in settings/base.py

**Comment by bufke** [comment]
@noliveleger can you review only the MFA related parts of this? At this time, only password based login is important. Oauth mostly works but will be finalized in a future PR.

**Review by JacquelineMorrissette — CHANGES_REQUESTED** [review]

**Comment by JacquelineMorrissette** [comment]
When setting up via kobo-install, if you don't setup STMP, it throws an error `[Open an interactive python shell in this frame] USError: [Errno 99] Cannot assign requested address` and I'm not sure that we want this to happen once it's merged in

File: hub/views.py [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
+#         """
+#         standard_fields = set(RegistrationForm().fields.keys())
+#         extra_fields = set(form.fields.keys()).difference(standard_fields)
+#         # Don't save the user unless we successfully store the extra data
+#         with transaction.atomic():
+#             new_user = super().register(form)
+#             extra_data = {k: form.cleaned_data[k] for k in extra_fields}
+#             new_user.extra_details.data.update(extra_data)
+#             new_user.extra_details.save()
+#         return new_user
```
[the remark this answers is not in the document]
Comment (bufke) [reply hub/views.py:26]: Forgot to implement! Fixed.

**Comment by bufke** [comment]
Regarding SMTP, is there a stacktrace for that?

**Comment by bufke** [comment]
@jnm do you think it makes sense to modify create_kobo_superuser to automatically verify the address? I'm going to implement it this way. Otherwise it's a barrier to getting up and running. Though it does reasonably error only after logging in and thus triggering the email, the install script is not the problem as far as I can tell.

**Comment by bufke** [comment]
Jackie will work on porting over the domain list work, which is currently failing tests.

**Comment by bufke** [comment]
Renamed https://github.com/kobotoolbox/kpi/pull/4163

**Pull request closed without merging**